.github/workflows/sync-upstream-push.yml

```diff
@@ -0,0 +1,53 @@
+name: Sync with Upstream (push)
+
+on:
+  pull_request_review:
+    types: [submitted]
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    if: github.event.review.state == 'approved' && github.event.pull_request.user.login == 'gha-runner-images-updater-test[bot]'
+    steps:
+      - name: Generate GitHub App token
+        id: generate-token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.GH_APP_ID }}
+          private-key: ${{ secrets.GH_APP_KEY }}
```

Comment:
Don't think this app is needed - `permissions` can do the job in the only place you're using the token.

(thinking slightly harder... ideally we would have a branch protection rule to prevent force pushes to `main`, and then you would need an app to be able to bypass that, but we're not doing that atm)

Reply:
You are right, not sure what led me to the GH App in the first place, it was something related to the workflow not being able to update other workflow files, but turns out it can? I'm not sure what changed.